File: BOJ/Graph/Silver/2644/2644.py
import sys
def dfs(cur,depth) :
    global ans
    visit[cur] = True
    if cur == b and ans == -1 : 
        ans = depth
    for next in nodes[cur] :
        if visit[next] : continue
        dfs(next,depth+1)
n = int(input())
a,b = map(int,input().split())
m = int(input())
nodes = [[] for _ in range(n+1)]
visit = [False for _ in range(n+1)]
for _ in range(m) :
    node = list(map(int,sys.stdin.readline().split()))
    nodes[node[0]].append(node[1])
    nodes[node[1]].append(node[0])
ans = -1
dfs(a,0)
print(ans)

File: BOJ/Graph/Silver/2644/2644.swift
func dfs(_ cur : Int,_ depth : Int) {
    visit[cur] = true
    if cur == b && ans == -1 {ans = depth}
    for next in nodes[cur] {
        if visit[next] {continue}
        dfs(next,depth+1)
    }
}
let n = Int(readLine()!)!
let input = readLine()!.split(separator: " ").map{Int(String($0))!}
let (a,b) = (input[0],input[1])
let m = Int(readLine()!)!
var nodes : [[Int]] = Array(repeating: [], count: n+1)
var visit : [Bool] = Array(repeating: false, count: n+1)
for _ in 0..<m {
    let node = readLine()!.split(separator: " ").map{Int(String($0))!}
    nodes[node[0]].append(node[1])
    nodes[node[1]].append(node[0])
}
var ans = -1
dfs(a,0)
print(ans)

